concent_api/core/payments/bankster.py

```diff
@@ -238,46 +201,47 @@ def finalize_payment(deposit_claim: DepositClaim) -> Optional[str]:
-            if subtask is not None:
-                task_to_compute = deserialize_message(subtask.task_to_compute.data.tobytes())
-                if task_to_compute.requestor_ethereum_address == deposit_claim.payer_deposit_account.ethereum_address:
-                    ethereum_transaction_hash = service.force_subtask_payment(  # pylint: disable=no-value-for-parameter
-                        requestor_eth_address=deposit_claim.payer_deposit_account.ethereum_address,
-                        provider_eth_address=deposit_claim.payee_ethereum_address,
-                        value=deposit_claim.amount,
-                        subtask_id=deposit_claim.subtask_id,
-                    )
-                elif task_to_compute.provider_ethereum_address == deposit_claim.payer_deposit_account.ethereum_address:
-                    ethereum_transaction_hash = service.cover_additional_verification_cost(  # pylint: disable=no-value-for-parameter
-                        provider_eth_address=deposit_claim.payer_deposit_account.ethereum_address,
-                        value=deposit_claim.amount,
-                        subtask_id=deposit_claim.subtask_id,
-                    )
-                else:
-                    assert False
-        else:
-            assert False
+    # If the DepositClaim still exists at this point, Bankster uses SCI to create an Ethereum transaction.
+    if deposit_claim.concent_use_case == ConcentUseCase.FORCED_ACCEPTANCE:
+        ethereum_transaction_hash = service.force_subtask_payment(  # pylint: disable=no-value-for-parameter
+            requestor_eth_address=deposit_claim.payer_deposit_account.ethereum_address,
+            provider_eth_address=deposit_claim.payee_ethereum_address,
+            value=deposit_claim.amount,
+            subtask_id=deposit_claim.subtask_id,
+        )
+    elif deposit_claim.concent_use_case == ConcentUseCase.ADDITIONAL_VERIFICATION:
+        subtask = Subtask.objects.filter(subtask_id=deposit_claim.subtask_id).first()  # pylint: disable=no-member
+        if subtask is not None:
+            task_to_compute = deserialize_message(subtask.task_to_compute.data.tobytes())
+            if task_to_compute.requestor_ethereum_address == deposit_claim.payer_deposit_account.ethereum_address:
+                ethereum_transaction_hash = service.force_subtask_payment(  # pylint: disable=no-value-for-parameter
+                    requestor_eth_address=deposit_claim.payer_deposit_account.ethereum_address,
+                    provider_eth_address=deposit_claim.payee_ethereum_address,
+                    value=deposit_claim.amount,
+                    subtask_id=deposit_claim.subtask_id,
+                )
+            elif task_to_compute.provider_ethereum_address == deposit_claim.payer_deposit_account.ethereum_address:
+                ethereum_transaction_hash = service.cover_additional_verification_cost(  # pylint: disable=no-value-for-parameter
```

Comment:
I know You didn't add this, just move indentation. But I just notice there's also no test for this case, and I wonder if there's even a case where provider needs to `cover_additional_verification_cost()`?

Reply:
If so, let's create an issue for that. I would like it to be merged ASAP, because there were some refactorings and I wouldn't like to stuck in hell of rebases... ;)